### DISCUSSION TITLE
Does the Instagram "spent request" count get reset every hour?

### DISCUSSION THREAD

**Discussion by Aston77** [AAndyProgram/SCrawler#126] [General]
Reading the official API limits it appears the call count is limited to 200 calls based on a 1hr rolling window.
https://developers.facebook.com/docs/graph-api/overview/rate-limiting#platform-rate-limits

In watching the spent call counter in the app I have seen it eventually reset, but it wasn't necessarily obvious what schedule it reset on.

Just out of curiosity, does it run on a 1hr rolling window in the background or is there a different logic that it operates off of?

![image](https://github.com/AAndyProgram/SCrawler/assets/78501124/39ad2388-8616-485c-bb0f-d0ff45186365)

**Comment by AAndyProgram** [comment]
Every 60 minutes
https://github.com/AAndyProgram/SCrawler/blob/b2a9b224780d969103ce7eebf541c49ec4253428/SCrawler/API/Instagram/SiteSettings.vb#L286-L291

**Reply by AAndyProgram** [reply]
Instagram has changed something again and the last month it hasn't been working very well. Almost every day it logs me out and forces me to update the credentials (x-csrf token & cookies).

**Reply by Aston77** [reply]
> Every 60 minutes
> 
> https://github.com/AAndyProgram/SCrawler/blob/b2a9b224780d969103ce7eebf541c49ec4253428/SCrawler/API/Instagram/SiteSettings.vb#L286-L291


I realize you've closed this and I don't have coding experience in .NET or C# so I wouldn't be qualified to submit this through more formal means, but given that, Instagram uses a rolling window for their API and this project appears to use a fixed window with a 60min reset once the limit has been hit, would there be any reason not to use a sliding window algorithm to more align with the API?

As I mentioned, I don't have the coding experience so I asked ChatGPT what such an implementation might look like 😁.

![image](https://github.com/AAndyProgram/SCrawler/assets/78501124/bbc2cf09-2618-4b9f-a5be-c81286599ba4)




> Instagram has changed something again and the last month it hasn't been working very well. Almost every day it logs me out and forces me to update the credentials (x-csrf token & cookies).

Yes, I have noticed that as well. I figured I had asked enough questions of you for the week though 😂

**Reply by AAndyProgram** [reply]
>I realize you've closed this

I closed this because I answered your question. But you can still continue the discussion if you want.

>Yes, I have noticed that as well. I figured I had asked enough questions of you for the week though 😂

You can ask as many questions as you like.

>Instagram uses a rolling window for their API and this project appears to use a fixed window with a 60min reset once the limit has been hit, would there be any reason not to use a sliding window algorithm to more align with the API?

Look, everything you described is only suitable for the API that can used through OAuth authorization. I don't use OAuth, so parsing this social network is much more difficult.

----

P.S. Why?! Why do people love this GPT shit so much? This is a thing to make people much dumber.

**Reply by Aston77** [reply]
> Look, everything you described is only suitable for the API that can used through OAuth authorization. I don't use OAuth, so parsing this social network is much more difficult.

Good to know, thanks for the context. I just noticed when I downloaded multiple timelines it would sit with a full green bar on the same job for more than an hour like it was frozen so I thought I would ask.
 
> P.S. Why?! Why do people love this GPT shit so much? This is a thing to make people much dumber.

I guess it depends on what you ask it. I've only been using it this last week but I find that search engines have gotten worse over the years to the point that its become tedious to piece together sparse information on less commonly discussed topics and ChatGPT (Bing in my case) does a pretty decent job of being able to contextually search, refine and search again on nuance questions that would take an immense amount of time to piece together otherwise.

In the above case, I was able to take your current format, have it adjusted relative to the Instagram guidelines and to our discussion, and pose a contextual question once I felt like it represented what I was trying to convey in a comprehensive enough way to not waste your time. 

Whereas for an otherwise casual question, I would have never asked it at all because it would have taken too long to pull the information together from traditional sources to adequately convey. Kind of like driving to a place you otherwise would never walk 🤷‍♂️.

**Reply by AAndyProgram** [reply]
>Good to know, thanks for the context. I just noticed when I downloaded multiple timelines it would sit with a full green bar on the same job for more than an hour like it was frozen so I thought I would ask.

It still doing something. My guess is that the profile you're downloading has too many posts. It takes some time. Update to the [new version](https://github.com/AAndyProgram/SCrawler/releases/latest). I've updated the progress bar to be more informative.

>but I find that search engines have gotten worse over the years to the point that its become tedious to piece together sparse information on less commonly discussed topics

But using the search engines, you can see different sources of information, you can analyze and choose the source that you trust. When you delegate the search to a program that thinks for you, you lose access to the truth! PLEASE REMEMBER THIS FOREVER! These bastards ask you to write your name, your phone, your e-mail just for registration! They closed the code of their system. Anything you ask for there is saved forever (allegedly for system learning purposes). Your documents, photos, videos, voice and all information that this chat can access will be stored forever and you will never be able to delete this data! Every scammer who breaks this system will know everything about you, and all you have to do is pray that the scammer only steals your money and not your identity!

And I haven't said anything yet about authorities and manipulating people's opinions!

**Reply by Aston77** [reply]
Oh I very much agree, for all the reasons you mentioned and then some. 

Can't say I feel particularly secure with the search engines either 😂.
